packages/authentication/src/strategy-adapter.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import * as http from 'http';
+import {ServerRequest as Request} from 'http';
+
+/**
+ * Interface definition of a passport strategy.
+ */
+export interface Strategy {
+  authenticate(req: http.ServerRequest): void;
+}
+
+/**
+ * Http Request interface
+ */
+export interface ParsedRequest extends Request {
+  // see http://expressjs.com/en/4x/api.html#req.path
+  path: string;
+  // see http://expressjs.com/en/4x/api.html#req.query
+  query: { [key: string]: string };
+  // see https://github.com/DefinitelyTyped/DefinitelyTyped/issues/15808
+  url: string;
+  pathname: string;
+  method: string;
+}
+
+/**
+ * Shimmed Request to satisfy express requirements of passport strategies.
+ */
+export class ShimRequest {
+  headers: Object;
+  query: Object;
+  url: string;
+  path: string;
+  method: string;
+  constructor(request?: ParsedRequest) {
+    if (request) {
+      this.headers = request.headers;
+      this.query = request.query;
+      this.url = request.url;
+      this.path = request.path;
+      this.method = request.method;
+    }
+  }
+}
+
+/**
+ * Adapter class to invoke a passport strategy.
+ * Instance is created by passing a passport strategy in the constructor
+ */
+export class StrategyAdapter {
+  private strategyCtor: Strategy;
+
+  constructor(strategy: Strategy) {
```

Comment:
I may be wrong but if you do

'''
Constructor(private strategy: Strategy) {
}
'''

Then you can drop the object variable strategyCtor and use strategy instead. The key point here is the private keyword which will do what your doing automatically, not a huge deal so feel free to ignore.

This probably explains it better https://www.stevefenton.co.uk/2013/04/stop-manually-assigning-typescript-constructor-parameters/